hack/tools/go.mod

```diff
@@ -5,74 +5,220 @@ go 1.19
 replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.4.8
 
 require (
+	github.com/aquasecurity/trivy v0.47.0
 	github.com/drone/envsubst/v2 v2.0.0-20210730161058-179042472c46
 	github.com/golang/mock v1.6.0
 	github.com/golangci/golangci-lint v1.52.2
 	github.com/itchyny/gojq v0.12.9
 	github.com/joelanford/go-apidiff v0.5.0
-	github.com/onsi/ginkgo/v2 v2.9.2
+	github.com/onsi/ginkgo/v2 v2.9.4
+	golang.org/x/vuln v1.0.1
 	gotest.tools/gotestsum v1.10.0
-	k8s.io/code-generator v0.26.10
+	k8s.io/code-generator v0.27.3
 	sigs.k8s.io/cluster-api/hack/tools v0.0.0-20231026175444-4ad3af14b736
 	sigs.k8s.io/controller-runtime/tools/setup-envtest v0.0.0-20211110210527-619e6b92dab9
 	sigs.k8s.io/controller-tools v0.11.4
-	sigs.k8s.io/kustomize/kustomize/v4 v4.5.7
+	sigs.k8s.io/kustomize/kustomize/v5 v5.0.4-0.20230601165947-6ce0bf390ce3
```

Comment:
and also a major version change here!

Reply:
for some reason `v4.5.x` kustomize build is failing causing smoke-test check failure, hence had bumped to v5.

Reply:
build of kustomize failing with latest v4, v5 seems to be working fine.